lang/cs/Org.Apache.REEF.IO.Tests/TestAzureDataLakeFileSystemE2E.cs

```diff
@@ -0,0 +1,229 @@
+using Microsoft.Azure.DataLake.Store;
+using Org.Apache.REEF.Tang.Interface;
+using System.IO;
+using System.Linq;
+
+namespace Org.Apache.REEF.IO.Tests
+{
+    /// <summary>
+    /// E2E tests for AzureDataLakeFileSystem.
+    /// These tests require the person running the test to fill in credentials.
+    /// </summary>
+    public sealed class TestAzureDataLakeFileSystemE2E : IDisposable
+    {
+        private const string SkipMessage = "Fill in credentials before running test"; // Use null to run tests
+        private const string ContentsText = "hello";
+        private IFileSystem _fileSystem;
+        private AdlsClient _adlsClient;
+        private string defaultFolderName;
+
+        public TestAzureDataLakeFileSystemE2E()
+        {
+            // Service principal / appplication authentication with client secret / key
+            // Use the client ID of an existing AAD "Web App" application.
+            // Fill in before running test!
+            const string adlsAccountName = "#####.azuredatalakestore.net";
+            const string tenant = "microsoft.onmicrosoft.com";
+            const string tokenAudience = @"https://datalake.azure.net/";
+            const string clientId = "#############################";
+            const string secretKey = "##########";
+
+            defaultFolderName = "reef-test-folder-" + Guid.NewGuid();
+
+            IConfiguration conf = AzureDataLakeFileSystemConfiguration.ConfigurationModule
+                .Set(AzureDataLakeFileSystemConfiguration.DataLakeStorageAccountName, adlsAccountName)
+                .Set(AzureDataLakeFileSystemConfiguration.Tenant, tenant)
+                .Set(AzureDataLakeFileSystemConfiguration.ClientId, clientId)
+                .Set(AzureDataLakeFileSystemConfiguration.SecretKey, secretKey)
+                .Build();
+
+            _fileSystem = TangFactory.GetTang().NewInjector(conf).GetInstance<AzureDataLakeFileSystem>();
+
+            ServiceClientCredentials adlCreds = GetCreds_SPI_SecretKey(tenant, new Uri(tokenAudience), clientId, secretKey);
+            _adlsClient = AdlsClient.CreateClient(adlsAccountName, adlCreds);
+        }
+
+        public void Dispose()
+        {
+            if (_adlsClient != null)
+            {
+                _adlsClient.DeleteRecursive($"/{defaultFolderName}");
+            }
+        }
+
+        [Fact(Skip = SkipMessage)]
+        public void TestOpenE2E()
+        {
+            string fileName = UploadFromString(ContentsText);
+            var stream = _fileSystem.Open(PathToFile(fileName));
+            StreamReader reader = new StreamReader(stream);
+            string streamText = reader.ReadToEnd();
+            Assert.Equal(ContentsText, streamText);
+        }
+
+        [Fact(Skip = SkipMessage)]
+        public void TestCreateE2E()
+        {
+            string fileName = $"/{defaultFolderName}/TestCreateE2E.txt";
+            var stream = _fileSystem.Create(PathToFile(fileName));
```

Comment:
Because this is end to end I would suggest checking the contents of the stream to make sure they match the expectation. Otherwise, you could be opening a stream to another file.

Reply:
This is an output stream to an empty file. Cannot read contents. The check is to make sure Create returns an outputstream to created file.